luigi/contrib/hdfs/format.py

```diff
@@ -75,9 +77,13 @@ def abort(self):
     def close(self):
         super(HdfsAtomicWriteDirPipe, self).close()
         try:
-            remove(self.path)
-        except HDFSCliError:
-            pass
+            if exists(self.path):
+                remove(self.path)
```

Comment:
Hello @dlstadther : 
This is the line that ties to remove a path without checking if it exists.
A looking at how it was implemented try/except/pass the spirit is: it it fails do not bother.
I added a check to verify if the path exists before removing it.